lib/tests/tests/diagnostics/return_in_closure.vrl

```diff
@@ -0,0 +1,29 @@
+# DIAGNOSTICS
+# result:
+#
+# error[E203]: syntax error
+# ┌─ :3:3
+# │
+# 3 │   return 42
+# │   ^^^^^^
+# │   │
+# │   unexpected syntax token: "Return"
```

Comment:
Is it correct that the only case we can encounter this error is if the `return` was inside a closure? If this is true, we can improve the error to avoid confusion.

Reply:
Agreed. I might suggest allowing it through the "lexer" but return a more specific error (`return cannot be used within closures`) when "compiling".